File: iOSPlugin/KarteSDKPlugin/KarteSDKPlugin/KartePlugin.swift
//
//  Dummy.swift
//  KarteSDKPlugin
//
//  Created by kota.fujiwara on 2020/09/10.
//  Copyright © 2020 PLAID, inc. All rights reserved.
//

import Foundation
import KarteCore

class KartePlugin: NSObject, Library {
    static var name: String {
        "unity"
    }
    
    static var version: String {
        "1.0.0"
    }
    
    static var isPublic: Bool {
        true
    }
    
    static func configure(app: KarteApp) {
    }
    
    static func unconfigure(app: KarteApp) {
    }
}

extension KartePluginLoader {
    open override class func handleLoad() {
        KarteApp.register(library: KartePlugin.self)
    }
}

File: androidPlugin/tracker_unity/src/main/java/io/karte/unity/KartePlugin.java
package io.karte.unity;
import io.karte.android.KarteApp;
import io.karte.android.core.library.Library;
import org.jetbrains.annotations.NotNull;

public class KartePlugin implements Library {
  // implements Library
  @Override
  public boolean isPublic() {
    return true;
  }

  @NotNull
  @Override
  public String getName() {
    return "unity";
  }

  @NotNull
  @Override
  public String getVersion() {
    return "1.0.3";
  }

  @Override
  public void configure(@NotNull KarteApp karteApp) {
  }

  @Override
  public void unconfigure(@NotNull KarteApp karteApp) {
  }
}
